front/pages/api/v1/w/[wId]/spaces/[spaceId]/apps/import.ts

```diff
@@ -0,0 +1,429 @@
+ *                                   description: Key of the schema entry
+ *                                 type:
+ *                                   type: string
+ *                                   description: Type of the schema entry
+ *                                 description:
+ *                                   type: string
+ *                                   description: Description of the schema entry
+ *                                   nullable: true
+ *     responses:
+ *       200:
+ *         description: Imported apps
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 apps:
+ *                   type: array
+ *                   items:
+ *                     type: object
+ *                     properties:
+ *                       id:
+ *                         type: integer
+ *                         description: Unique identifier for the app
+ *                       sId:
+ *                         type: string
+ *                         description: Unique string identifier for the app
+ *                       name:
+ *                         type: string
+ *                         description: Name of the app
+ *                       description:
+ *                         type: string
+ *                         description: Description of the app
+ *                       savedSpecification:
+ *                         type: string
+ *                         description: Saved specification of the app
+ *                       savedConfig:
+ *                         type: string
+ *                         description: Saved configuration of the app
+ *                       savedRun:
+ *                         type: string
+ *                         description: Saved run identifier of the app
+ *                       dustAPIProjectId:
+ *                         type: string
+ *                         description: ID of the associated Dust API project
+ *       400:
+ *         description: Bad Request. Missing or invalid parameters.
+ *       401:
+ *         description: Unauthorized. Invalid or missing authentication token.
+ *       404:
+ *         description: Workspace not found.
+ *       405:
+ *         description: Method not supported.
+ *       500:
+ *         description: Internal Server Error.
+ */
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<GetAppsResponseType>>,
+  auth: Authenticator,
+  { space }: { space: SpaceResource }
+): Promise<void> {
+  if (!auth.isSystemKey()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "invalid_oauth_token_error",
+        message: "Only system keys are allowed to use this endpoint.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST":
+      const r = PostAppsRequestSchema.safeParse(req.body);
```

Comment:
Can we instead make it a workflow there is no need for an API no?

Reply:
The idea was to have a github workflow that would fetch apps and push them to another domain through api - do you think of a temporal workflow that would itself do the fetch on US and import apps ?

Reply:
I see the value of a github action as it's aligned with our current tooling. My main FUD is the existence of that write endpoint (even if protected by system key). How about a special endpoint under poke that does not take any input but does the work. Mostly reduce the semantic to "please sync" vs "override with this"

Reply:
I moved the code to a full sync triggered in poke ("just sync") - it could also be added in a croned temporal workflow, to automatically sync every 10 mins for example.

Reply:
SGTM